File: Data Structures & Algorithms/maximum-subarray/submission-1.go
func maxSubArray(nums []int) int {
    res, curSum := nums[0], nums[0]
    n := len(nums)
    for i := 1; i < n; i++ {
        curSum += nums[i]
        if curSum < nums[i] {
            curSum = nums[i]
        }
        res = max(res, curSum)
    }
    return res
}

File: Data Structures & Algorithms/maximum-twin-sum-of-a-linked-list/submission-1.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def pairSum(self, head: Optional[ListNode]) -> int:
        # Collect values into an array — one pass, then O(1) indexed access
        vals = []
        cur = head
        while cur:
            vals.append(cur.val)
            cur = cur.next
        
        n = len(vals)
        res = 0
        for i in range(n // 2):
            res = max(res, vals[i] + vals[n - 1 - i])
        return res